libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationViewModel.java

```diff
@@ -403,6 +405,7 @@ private LocationEngine initializeLocationEngineFrom(final NavigationViewOptions
       ReplayLocationEngine replayLocationEngine = new ReplayLocationEngine(mapboxReplayer);
       final Point lastLocation = getOriginOfRoute(options.directionsRoute());
       ReplayEventBase replayEventOrigin = ReplayRouteMapper.mapToUpdateLocation(0.0, lastLocation);
+      mapboxReplayer.registerObserver(replayLocationEngine);
```

Comment:
should we remove this? I think we already `registerObserver` when create `ReplayLocationEngine`: https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/ReplayLocationEngine.kt#L29

```kotlin
class ReplayLocationEngine(
    mapboxReplayer: MapboxReplayer
) : LocationEngine, ReplayEventsObserver {

    private val registeredCallbacks: MutableList<EngineCallback> = mutableListOf()
    private val lastLocationCallbacks: MutableList<EngineCallback> = mutableListOf()
    private var lastLocationEngineResult: LocationEngineResult? = null

    init {
        mapboxReplayer.registerObserver(this)
    }
```

Reply:
oh good catch, that change comes in a future pr
https://github.com/mapbox/mapbox-navigation-android/pull/3191/files#diff-a006a578550421e51b5424f39e5fbc95R408